src/components/bottom_bar_statistic/aggregate_statistics_store.ts

```diff
@@ -135,10 +136,10 @@ export class AggregateStatisticsStore extends SpreadsheetStore {
       // does not match the data handled by the function.
       // Ex: if there are only texts in the selection, we prefer that the SUM result
       // be displayed as undefined rather than 0.
-      let fnResult: number | undefined = undefined;
+      let fnResult: Lazy<number> | undefined = undefined;
       const evaluatedCells = getCells(fn.types.join(","));
       if (evaluatedCells.length) {
-        fnResult = fn.compute(evaluatedCells, locale);
+        fnResult = lazy(() => fn.compute(evaluatedCells, locale));
```

Comment:
we could even make the cell iteration and filter part of the lazy computation
```ts
    const fnResult: Lazy<number| undefined> = lazy(() => {
        const evaluatedCells = getCells(fn.types.join(","));
        if (evaluatedCells.length) {
          return fn.compute(evaluatedCells, locale);
        }
        return undefined;
      });
```

Reply:
no we cannot. We check the presence of fnResult in the final object to know if we should display the statistics button or not. So we'd end up evaluating everything all the time